### DISCUSSION TITLE
source object with insert_pdf

### DISCUSSION THREAD

**Discussion by fareshan** [pymupdf/PyMuPDF#3812] [Looking for help] [answered]
I have this error
```
code=4: source object number out of range
```
with `insert_pdf`.

Inspired by https://github.com/pymupdf/PyMuPDF/issues/856, I got a workaround: use `.tobytes(garbage=1)` before `insert_pdf`
```
source_doc.tobytes(garbage=1)
doc.insert_pdf(source_doc, from_page=0, to_page=source_doc.page_count - 1, start_at=-1)
```

But I don't really need source_doc to be converted to bytes. So I don't feel this is the right way to handle the issue. Do you know a better solution ?

**Comment by JorjMcKie** [comment, marked as answer]
Outputting to some target (`tobytes()` is actually equivalent to `save()`) obviously was sufficient (caused by your garbage option) to encounter and correct the internal error.

In some way this had to happen in some similar way. If you feel better you could have reopened the doc using that bytes output 😉. Maybe also more stable for other cases ...

**Reply by fareshan** [reply]
I agree. Reopening the doc using that bytes output makes more sense. Thanks!
